internal/release/promote/promotion.go

```diff
@@ -126,27 +126,34 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 		return "", fmt.Errorf("previewing: %w", err)
 	}
 
-	confirmed, err := p.promptProvider.ConfirmCreatingPromotionPullRequest()
+	answer, err := p.promptProvider.SelectCreatingPromotionPullRequest()
 	if err != nil {
-		return "", fmt.Errorf("confirming creating promotion pull request: %w", err)
-	}
-	if !confirmed {
-		p.promptProvider.PrintCanceled()
-		return "", nil
+		return "", fmt.Errorf("selecting create promotion pull request: %w", err)
 	}
 
-	if opts.TargetEnv.Spec.Promotion.AllowAutoMerge && !opts.AutoMerge {
-		autoMerge, err := p.promptProvider.ConfirmAutoMergePullRequest()
-		if err != nil {
-			return "", fmt.Errorf("confirming automerge: %w", err)
+	var performParams = PerformParams{
+		list: list,
+	}
+	switch answer {
+	case Draft:
+		performParams.draft = true
+		break
+	case Ready:
+		if opts.TargetEnv.Spec.Promotion.AllowAutoMerge && !opts.AutoMerge {
+			autoMerge, err := p.promptProvider.ConfirmAutoMergePullRequest()
+			if err != nil {
+				return "", fmt.Errorf("confirming automerge: %w", err)
+			}
+			opts.AutoMerge = autoMerge
```

Comment:
here you do not need to modify opts.AutoMerge since you are modifying performParams.autoMerge.

I modified opts before, so that I can use that value when declaring the PerformParams.